packages/store/src/index.ts

```diff
@@ -0,0 +1,231 @@
+import type { SetStoreFunction } from "solid-js/store";
+import { createStore } from "solid-js/store";
+
+type AnyFunction = (...args: any[]) => any;
+type Action<A extends AnyFunction> = (...args: Parameters<A>) => ReturnType<A>;
+type CreateAction = <A extends AnyFunction>(fn: A) => Action<A>;
+type CreateActions = <A extends object | undefined>(fns?: A) => A | undefined;
+
+export const createAction: CreateAction =
+  fn =>
+  (...args) =>
+    batch(() => untrack(() => fn(...args)));
+export const createActions: CreateActions = actions => {
+  if (!actions) return undefined;
+
+  let result = { ...actions };
+  // for (const [name, fn] of Object.entries(result) as [keyof typeof result, AnyFunction][]) {
+  for (const [name, fn] of Object.entries(result)) {
+    if (typeof fn !== "function") continue;
+    result = {
+      ...result,
+      [name]: createAction(fn)
+    };
+  }
+
+  return result;
+};
+
+/**
+ * Create a shared store.
+ *
+ * @typedef SharedStore
+ * @template <T: state type of the store, G: store access methods, A: store mutation methods>
+ * @returns {
+ *  {T} state - The current state of the store.
+ *  {G} getters - Predefined methods to access the state of the store.
+ *  {A} actions - Predefined methods to mutate the state of the store.
+ *  {SetStoreFunction<T>} setState - A function to manually mutate the current state of the store.
+ * }
+ */
+export type SharedStore<T extends object, G = undefined, A = undefined> = {
+  state: T;
+  getters: undefined extends G ? undefined : G;
+  actions: undefined extends A ? undefined : A;
+  setState: SetStoreFunction<T>;
+};
+
+/**
+ * Factory to create reusable stores.
+ *
+ * @typedef StoreFactory
+ * @template <T: state type of the store, G: store access methods, A: store mutation methods>
+ * @returns (initialValueSetter?: T | ((fallbackState: T) => T)) => SharedStore<T, G, A>
+ */
+export type StoreFactory<T extends object, G = undefined, A = undefined> = (
+  initialValueSetter?: T | ((fallbackState: T) => T)
+) => SharedStore<T, G, A>;
+
+/**
+ * createSharedStore takes an `initialState` and an optional object containing functions to create getters and/or actions.
+ * A store is created and returned along with access and mutation functions & the native `setState`.
+ * @template <T: state type of the store, G: store access methods, A: store mutation methods>
+ * @param initialState object to be wrapped in a Store
+ * @param createMethods optional object containing functions to create getters and/or actions.
+ * @returns `SharedStore`
+ * @example
+ * ```tsx
+ * // `counter-store.ts`
+ * export const counterStore = createSharedStore({
+ *   value: 5,
+ * }, {
+ *  getters: state => ({
+ *   count: () => state.value,
+ *  }),
+ *  actions: (setState, state) => ({
+ *   increment: () => setState(val => ({ ...val, value: val.value + 1 })),
+ *   reset: () => setState("value", 0),
+ *  })
+ * });
+ *
+ * // `Example.tsx`
+ * import { counterStore } from './counter-store.ts';
```

Comment:
Separation into multiple files is unnecessary information to add to the documentation in my opinion. It's not like this is the required way to use the primitive.